### DISCUSSION TITLE
Thinking about Organization

### DISCUSSION THREAD

**Discussion by sigrunixia** [obsidian-community/web-clipper-templates#1] [General]
We just started, so this is not set in stone yet, but organization is likely going to need to be thought about from day one. 

Sort:
By function?
By site?
By category?

What are your thoughts?

**Comment by esodesod** [comment]
Personally, I think **prefix** with site name (based on URL) makes sense, as the **URL** will (usually) be unique, and the main "template trigger". Might be a combination with schema.org (that I'm not using), but the URL will always be unique, hence easy for users to understand what (and where) the template is to be used?

Example template triggers I've use for GitHub, is e.g.
github.com-issues uses trigger like `/^https.+github.+issues.+$/`
github.com-releases uses trigger like `/^https.+github.+releases.+$/`

Example on (only) prefixed template names, with some info/type:
reddit.com-post
reddit.com-post-with-images
reddit.com-post-with-comments
github.com-issues
github.com-releases
... etc.

We could possibly add some kind of **suffix** with multiple variants of same template, or maybe just add author username, or something - mainly to differentiate on "duplicate" site templates, but with different variants/preferences? IDK - kinda depends if we want different "ready templates" and/or just showcase what's possible.

For **suffix**, e.g. if we have multiple "conflicting" (variations/personal preference on template), we could do something like:
github.com-releases-**esod** (example, based on `esod`s my preferences)
github.com-releases-**sigrunixia** (example, based on `sigrunixia`'s preferences)

Benefit will be easy to add similar templates, but with a personal touch, though keeping the prefix.

**Reply by sigrunixia** [reply]
Are there any that should be grouped by folders at the start?

**Reply by sigrunixia** [reply]
Thinking more, yes, they should. It's highly unlikely we'll run into the Github 1000 file/ 1000 folder limit.

**Reply by DudeThatsErin** [reply]
You could also do separate folders for each site... so reddit ones could be:
```
templates/reddit/post
templates/reddit/post-with-images
templates/reddit/post-with-comments
```
Github would have its separate folder.

Just to break it up even more and then on the forum, server or subreddit if people are looking for specific sites for templates, those folders can be linked.

**Reply by esodesod** [reply]
> Just to break it up even more and then on the forum, server or subreddit if people are looking for specific sites for templates, those folders can be linked.

Good point; I see the value using folders, **if** a folder will be the main entry point for a user, for site-specific templates.

**Reply by esodesod** [reply]
I also realized that the "default file name" when exporting the template from the clipper extension will be prefixed somewhat with the words of the actual "template name", e.g. a template named "YouTube with transcript" will result in file name `youtube-with-transcript-clipper.json`, which possibly adds some new ideas/considerations:

- using a good prefix in the file name (e.g., site-specific/based on the site domain name) is still a good practice, as the user will likely not have a template (only) named "post with images" but rather something like "reddit post with images", etc., which again will have transferred value for the default export file name.
- if the template name includes a good prefix (by default), do folders improve searchability? If so, how?
- if someone updates a template in the clipper, and simply wishes to update the community template, it's quick & easy (IMO) to export > last used folder (one place) > git diff/commit/push, rather than first navigating corresponding subfolder, and possibly an additional search & rename operation (user have to know what template to update and then rename the file correspondingly, though that's a valid eighter way...).

Another case for a good file name (in the first place), will be that if a user downloads a template (json), the (downloaded) template name will be easier to match if the file name is "good by default", e.g. `post-with-images-clipper.json` is "bad", as it could be a post for any site, e.g. both reddit and twitter, versus a prefixed name, like `reddit-post-with-images-clipper.json` and `twitter-post-with-images-clipper.json`, which is both unique based on site/domain, and gives more context to what the file actually is, and where it applies.
 
I guess it's just (my) personal preferences triggering some added feedback here (sorry 😅), but to me, it still feels easier to just go into the templates folder, then simply search for whatever I need (one place), rather than browsing multiple subfolders and/or search, spanning results from multiple subfolders.

Maybe I'm overthinking. I ironically am trying to reduce friction (in both adding, maintaining, and searching the templates). Everything is doable. In the end, I'm using local fuzzy searching, combining both file name and regex in-file-matching, so it really shouldn't matter that much to me 😆.

**Reply by sigrunixia** [reply]
With the announcement to Omnivore today, I suspect we'll need to move faster than expected.

Maybe.. until we get the demand to warrant the extended sorting, we should go with 1 folder for the templates themselves, the json files.

It's for Obsidian Publish that we cannot use one giant list, but we have more navigation tools there, and I can open a separate issue for that. #6

**Reply by DudeThatsErin** [reply]
@sigrunixia I agree. Shocking announcement.

@esodesod I didn't think about that, you are right... it would end up being `reddit/reddit-post-with-images.json` which is annoying to read and everything.

I think 1 folder (for now) is fine and *maybe* in the future switch to author sub-folder names or something like that.

**Comment by sigrunixia** [comment]
I think I will start on the PR side first. This has *some* format with #6 

What we need:
- The template (.json)
- The template guide (.md)
- Ideally, a screenshot that shows the template post import. It doesn't need to be in default theme. (img, which we can make into a .webp).

This means we need a system that helps:
1. Keep track of the 3 needs
2. Reduces submission and processing repetition
3. Makes these easy to update

**Comment by Purintu** [comment]
Hi can you add please more clarification on how to use the template of Youtube timestamps ? I can't figure out how do we add that timestamps in an existing note
